### PR TITLE
resource depends on building

### PR DESCRIPTION
**Pull request by ValeriyaBugakovaMT**

### CODE REVIEW

File: GrindingCity.WebApi/Controllers/BuildingController.cs [hunk 1]
```diff
@@ -13,48 +14,56 @@ public class BuildingController : ControllerBase
...
         // GET all api
         [HttpGet()]
-        public async Task<ActionResult<IEnumerable<Building>>> GetAllBuildings(
-            [FromServices] IBuildingRepository handler)
+        public async Task<ActionResult<GetAllBuildingsResponse>> GetAllBuildings(
+            [FromServices] IBuildingRepository buildingHandler)
         {
-            var result = await handler.GetAllBuildings();
+            var buildings = await buildingHandler.GetAllBuildingsAsync();
+            var result = new GetAllBuildingsResponse(buildings.Select(_ => _.Id));
```
Comment (F0urth) [inline GrindingCity.WebApi/Controllers/BuildingController.cs]: We shouldn't use discard operator like this (`_`)

File: GrindingCity.WebApi/Controllers/BuildingController.cs [hunk 2]
```diff
@@ -13,48 +14,56 @@ public class BuildingController : ControllerBase
...
 
             return result is null ? NotFound() : Ok(result);
```
Comment (F0urth) [inline GrindingCity.WebApi/Controllers/BuildingController.cs:24]: Is result ever null if we create object line above?

File: GrindingCity.WebApi/Controllers/BuildingController.cs [hunk 3]
```diff
@@ -13,48 +14,56 @@ public class BuildingController : ControllerBase
...
+            [FromServices] IResourceRepository resourceHandler)
         {
-            var result = await handler.GetBuilding(id);
+            var building = await buildingHandler.GetBuildingAsync(id) ?? throw new InvalidBuidingException();
+            var resources = await resourceHandler.GetAllResourcesAsync();
+            var result = new GetBuildingResponse(
+                        building.Id, 
+                        building.Price, 
+                        building.Type, 
+                        resources.Where(_ => _.BuildingId == building.Id));
```
Comment (F0urth) [inline GrindingCity.WebApi/Controllers/BuildingController.cs]: Same as before we shouldn't use discard operator like this

File: GrindingCity.WebApi/Controllers/ResourseController.cs [hunk 4]
```diff
@@ -7,52 +8,60 @@ namespace GrindingCity.WebApi.Controllers
...
         [HttpGet()]
-        public async Task<ActionResult<IEnumerable<Resourse>>> GetAllResourses(
-            [FromServices] IResourseRepository handler)
+        public async Task<ActionResult<IEnumerable<Resource>>> GetAllResources(
+            [FromServices] IResourceRepository resourceHandler)
         {
-            var result = await handler.GetAllResourses();
+            var result = await resourceHandler.GetAllResourcesAsync();
 
             return result is null ? NotFound() : Ok(result);
```
Comment (F0urth) [inline GrindingCity.WebApi/Controllers/ResourseController.cs]: do we ever gonna have null there?

File: GrindingCity.WebApi/Controllers/ResourseController.cs [hunk 5]
```diff
@@ -7,52 +8,60 @@ namespace GrindingCity.WebApi.Controllers
...
     [Route("api/[controller]")]
     [ApiController]
-    public class ResourseController : Controller
+    public class ResourceController : Controller
     {
         // GET all api
         [HttpGet()]
-        public async Task<ActionResult<IEnumerable<Resourse>>> GetAllResourses(
-            [FromServices] IResourseRepository handler)
+        public async Task<ActionResult<IEnumerable<Resource>>> GetAllResources(
```
Comment (F0urth) [inline GrindingCity.WebApi/Controllers/ResourseController.cs]: Exposion of domain we should not do this

File: GrindingCity.WebApi/Controllers/ResourseController.cs [hunk 6]
```diff
@@ -7,52 +8,60 @@ namespace GrindingCity.WebApi.Controllers
...
 
             return result is null ? NotFound() : Ok(result);
         }
 
         // GET api
         [HttpGet("{id:guid}")]
-        public async Task<ActionResult<Resourse>> GetResourse(Guid id,
-            [FromServices] IResourseRepository handler)
+        public async Task<ActionResult<Resource>> GetResource(Guid id,
+            [FromServices] IResourceRepository resourceHandler)
```
Comment (F0urth) [inline GrindingCity.WebApi/Controllers/ResourseController.cs]: we should name repositories as repositories they are not handlers

File: GrindingCity.WebApi/Controllers/ResourseController.cs [hunk 7]
```diff
@@ -7,52 +8,60 @@ namespace GrindingCity.WebApi.Controllers
...
         // PUT api
         [HttpPut("{id:guid}")]
-        public async Task<ActionResult> UpdateResourse(Guid id, [FromBody] UpdateResourseRequest resourse,
-            [FromServices] IResourseRepository handler)
+        public async Task<ActionResult> UpdateResource(Guid id, [FromBody] UpdateResourceRequest request,
+            [FromServices] IResourceRepository resourceHandler,
+            [FromServices] IBuildingRepository buildingHandler)
         {
-            await handler.UpdateResourse(id, resourse);
+            _ = await buildingHandler.GetBuildingAsync(request.buildingId)
```
Comment (F0urth) [inline GrindingCity.WebApi/Controllers/ResourseController.cs]: why don't we handle this with http codes?

File: GrindingCity.WebApi/Controllers/BuildingController.cs (hunk 1 above)
Comment (ValeriyaBugakovaMT) [reply GrindingCity.WebApi/Controllers/BuildingController.cs]: done

File: GrindingCity.WebApi/Controllers/ResourseController.cs (hunk 7 above)
Comment (ValeriyaBugakovaMT) [reply GrindingCity.WebApi/Controllers/ResourseController.cs]: updated, like this if (building is null) return NotFound();?

**Pull request merged**